src/queries/full-text-queries/query-string-query.js

```diff
@@ -303,6 +309,42 @@ class QueryStringQuery extends QueryStringQueryBase {
         this._queryOpts.escape = enable;
         return this;
     }
+
+    /**
+     * Sets the type of query string query. Valid values are:
+     * - `best_fields` - (default) Finds documents which match any field,
+     * but uses the `_score` from the best field.
+     *
+     * - `most_fields` - Finds documents which match any field and combines
+     * the `_score` from each field.
+     *
+     * - `cross_fields` - Treats fields with the same `analyzer` as though
+     * they were one big field. Looks for each word in *any* field
+     *
+     * - `phrase` - Runs a `match_phrase` query on each field and combines
+     * the `_score` from each field.
+     *
+     * - `phrase_prefix` - Runs a `match_phrase_prefix` query on each field
+     * and combines the `_score` from each field.
+     *
+     * - `bool_prefix` - Creates a match_bool_prefix query on each field and
+     * combines the _score from each field.
+     *
+     *
+     * @param {string} type Can be one of `best_fields`, `most_fields`,
+     * `cross_fields`, `phrase`, `phrase_prefix` and `bool_prefix`. Default is
+     * `best_fields`.
+     * @returns {QueryStringQuery} returns `this` so that calls can be chained.
+     */
+    type(type) {
```

Comment:
As far as I can see, this option was added only in Elasticsearch v6 based on the [docs for v5](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-dsl-query-string-query.html#query-string-multi-field-parms). Can we add a note calling out the same?